### DISCUSSION TITLE
MethodExpr or ##func() syntax documentation

### DISCUSSION THREAD

**Discussion by chwmilcon** [kevinmehall/rust-peg#389] [General]
I found syntax within the `grammar.rustpeg` that appears to be means of calling an external parsing function within the rust-peg grammar. `rule PAREN_GROUP()   -> Group = ##group(Delimiter::Parenthesis)`

This is a great feature, but I'm not finding documentation for it for rust-peg. Am I missing it?

**Comment by kevinmehall** [comment]
It's a hack that's there because I needed it for the PEG meta-grammar, but I never documented it, because I'd planned to replace it with something more flexible not tied to the input type: https://github.com/kevinmehall/rust-peg/issues/284

Not going to break it without a breaking semver bump, so feel free to use it though.

**Comment by chwmilcon** [comment]
Thank you for clarifying. It feels similar to the `@extern(...)` from peginator
